plugins/shuffle/round.py

```diff
@@ -215,6 +217,16 @@ def process_announcement(self):
                     self.log_message("encrypt new address")
                     self.phase = 'BroadcastOutput'
 
+    def _parse_from_string_safe(self, msg, sender):
+        ''' Parses messages from 'string' msg (really bytes).
+        Returns sender if there is a decode error, otherwise returns None '''
+        try:
+            self.messages.packets.ParseFromString(msg)
```

Comment:
Is this implemented in protobuf/text_format.py ? Or somewhere else?
This will work to avoid errors. If the right message never shows up, I assume the caller of this chain will time out and give up.

Reply:
Yeah it's a protobuf function that raises on badly formatted messages.  he never bothered to ever check for it, hence the need for this.

Yeah ultimately either there will be an immediate abort-with-blame or in some cases timeout.

The code is not very resilient to bad/garbage messages.  Likely there are still extant holes.  Just trying to plug up what I can. Bad actors shouldn't be causing clients to have exception crash reporter windows pop up willy nilly.  That's just silly.